### DISCUSSION TITLE
complete explosion overhaul

### DISCUSSION THREAD

**Discussion by LePiromano** [FakeFishGames/Barotrauma#8144] [Ideas and suggestions]
I put it up to community vote to change entirely how explosions work, this is my suggestion but you can add other ideas as a comment.

Part 1: "Solidify" damage
Currently, according to devs "the damage values you see on the code are just a theoretical value explosions could do to a target's limbs were all close to the center of the explosion." I consider this too complex, unless there's a reason behind why it is that way. I think explosions should work as a "propagating impact damage", which would make damage values accurate to the code and easy to understand.

Part 2: Divide damage per limb, not multiply
Explosions right now are weak in terms of damage as mentioned in part 1, but they make up for it by multiplying their damage on every limb the explosion reaches. This sounds ok, but it is very inconsistent, for example mudraptors take 27~ times the damage of an explosion (if every limb is reached by it), while leucocytes only take the explosion damage once since they only have 1 limb. If part 1 and 2 are added, explosions would already be way cleaner and consistent.

Part 3: Damage falloff
Damage falloff of explosions is almost exaggerated right now, explosions can barely be used as aoe damage weapons because of it. The change i suggest is simple, make damage fall off go from 100% to 50% damage, not 100% to 0%.

Part 4: Weakspot concentration
This is more of a personal idea, when an explosive projectile directly hits a weakspot (eg: moloch brain, charybdis mouth), the damage wont disperse and the weakspot limb will take all damage, the rest of the explosion would still affect other creatures around, of course.

Please share your opinions, mostly if you're in the design team :D

**Comment by Regalis11** [comment]
Thank you for the ideas! :) Even though I agree regarding the issues in the current damage logic, I'm not entirely convinced about the proposed solutions.

> I consider this too complex, unless there's a reason behind why it is that way

I'd say the reason for this is to make the explosions more realistic (e.g. the distance to the explosion and which parts of your body are exposed to the explosion affects how much damage you take) and to differentiate them from "normal" damage.

> Divide damage per limb, not multiply

As I mentioned in the ticket, I don't think the damage should be divided per limb. That'd basically invert the issue, making monster with more limbs take less damage from the explosions. I think dividing the damage by the number of limbs the explosion hits could work though.

> The change i suggest is simple, make damage fall off go from 100% to 50% damage, not 100% to 0%.

I agree that we should change how the falloff is calculated, but idk if that'd be the best way to go about it. To me it'd feel strange (and unrealistic?) that an explosion could kill someone standing 100 meters from it, and leave someone 101 meters away completely unharmed. Perhaps the damage could decrease non-linearly instead (e.g. square root of the distance)?

But in any case, we'll take these ideas into consideration when we get around to overhauling the explosions!

**Reply by LePiromano** [reply]
Thanks for considering it! Remember at the end what the community seeks is a fun game, even if it kills some of the realism :D

**Comment by Udrakan** [comment]
What I don't like about explosives atm is how the damage is so unpredictable. I think general improvements to the limb damage and falloff will remove some of this uncertainty and make them feel better.

**Reply by Ek-Videogames** [reply]
- As stated in the respective ticket, some monsters use 'limb-based' armor and others use flat damage reductions.

- Different explosions sometimes use vastly different affliction types, between lacerations, burns, or tissue injury.

- Some monsters have much more different affliction resistances which - due to item 2 - can make the damage of any given weapon very inconsistent between enemies. I respect the concept of having some enemies weaker to certain damage types, but the difference should not be as extreme as the recent issues with Pulse Laser or Incendiary Shotgun Slugs being ineffective against enemies with burn resistance or humans with burn resistant clothes.

Part of the solution is tweaking how explosions work, but part of the solution is also tightening up some of the design inconsistencies like weapons & creatures having outdated or nonsensical stats.

**Comment by ghost** [comment]
Base damage multiplication per thing hit bears little to no sense, honestly. It's indeed quite illogical.
Unless water on Europa is significantly different from water we use in our daily lives, damage falloff in it won't be anywhere close to 100% in center > 0% at edge.
Even disregarding real-life physics, it's... not entirely OK to have blastwave dealing noticeable damage in 2/3 of its range anyway. People won't understand.

Since Barotrauma aint Silent Hunter 6, I'd suggest keeping it simple and arcade'ish.
Thing hit directly always takes 100% of explosion's base damage (in addition to any damage from impact itself). Anything else in range takes... let's say, 50% of explosion's base damage with no falloff whatsoever.
Things slightly outside of explosion's range (let's say, which avoided damage somehow but were still present within 110% explosion's range) are stunned for 0.5-2 seconds (depending on explosion's base damage vs BASE explosion range, the higher the former and the lower the latter - the longer the stun) and SLIGHTLY propelled outwards from impact point.
Optionally, solids (crew's handheld shields (if big enough, if stout enough), monsters' exceptionally hard carapaces, sub hulls) protect anything non-solid behind them from damage - but (!) not from resulting stun.
And everyone's happy!
No complicated coding, no edge cases, no "hey, why it din't affected that dude?!" moments, viability vs EVERYTHING from Endworms to Clowns.

From my experience with World of Tanks, the more *beep* the developer introduces into explosion calculations - the more confusion it causes for end users (players), the more *beep*y variables to balance, the more *beep*ing unforeseen consequences from seemingly innocent balance changes...
